hs_core/views/__init__.py

```diff
@@ -697,3 +697,18 @@ def _set_resource_sharing_status(request, user, resource, is_public, is_discover
         # set isPublic metadata AVU accordingly
         istorage = IrodsStorage()
         istorage.setAVU(resource.short_id, "isPublic", str(resource.raccess.public))
+
+
+def _get_message_for_setting_resource_flag(resource, resource_flag):
```

Comment:
Not sure what changes you are suggesting here. The existing 'can_be_public' property' is a thin wrapper that uses 'has_all_required_elements()' and 'has_required_content_files(). Are you suggesting that we delete the 'can_be_public' property?  I would prefer to have this helper function.  But changing the name of this property to 'can_be_public_or_discoverable' is a good idea.